### PR TITLE
Initial documentation for Windows Deployment

### PR DESCRIPTION
**Pull request by VeArnold**
_Description of what this PR is changing or adding, and why:_

The PR adds an initial Windows deployment page for Flutter Windows Desktop applications, which gives a brief overview of deployment to Microsoft Store. Similar pages already exist for macOS and Linux Desktop applications and can be expanded on.

Relevant issues:
https://github.com/flutter/website/issues/6862
https://github.com/flutter/website/issues/6466
https://github.com/flutter/website/issues/6465

### CODE REVIEW

**Comment by domesticmouse** [comment]
Please sign the CLA

File: src/deployment/windows.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+Additional options (such as the list of capabilites and language resources 
+contained in the package) can be configured using the aforementioned package.
+
+For publishing, Codemagic uses the [Partner Center submission API][partnercenterapi]; thus,
+Codemagic requires [associating the Azure Active Directory and Partner Center accounts][azureadassociation].
+
+## Updating the app's version number
+
+With Flutter Windows Desktop, the version number must be set during the 
+packaging process and can not be set via the `pubspec.yaml`
```
Comment (domesticmouse) [inline src/deployment/windows.md:81]: We should probably document how to set this here.

Context: https://github.com/flutter/website/issues/6466

File: src/deployment/windows.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+For Flutter applications, either the [Codemagic Workflow Editor][cmworkfloweditor] or [codemagic.yaml][cmyaml] 
+can be used to package the application and deploy to Microsoft Partner Center.
+Additional options (such as the list of capabilites and language resources 
+contained in the package) can be configured using the aforementioned package.
+
+For publishing, Codemagic uses the [Partner Center submission API][partnercenterapi]; thus,
+Codemagic requires [associating the Azure Active Directory and Partner Center accounts][azureadassociation].
+
+## Updating the app's version number
```
Comment (domesticmouse) [inline src/deployment/windows.md:78]: Please add a section for setting an application's name.

Context: https://github.com/flutter/website/issues/6465

**Review by domesticmouse — COMMENTED** [review]
PTAL @sfshaza2

**Review by sfshaza2 — CHANGES_REQUESTED** [review]
@blasten, @VeArnold has very kindly written this new page describing how to deploy an app to the MS store. Our team (@domesticmouse) has reviewed it and made a few suggestions, but could you cast your eyes over it as well?

@VeArnold, thanks so much for doing this! Could you also add this page to the side nav? The affected file is src/_data/sidenav.yml. Thx!!

**Review by domesticmouse — APPROVED** [review]

**Review by sfshaza2 — APPROVED** [review]
lgtm!

**Pull request merged**